torchgeo/datamodules/vhr10.py

```diff
@@ -0,0 +1,185 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""NWPU VHR-10 datamodule."""
+
+from typing import Any, Callable, Dict, List, Tuple, Union
+
+import kornia.augmentation as K
+import torch
+import torchvision
+from einops import rearrange
+from torch import Generator, Tensor
+from torch.nn.modules import Module
+from torch.utils.data import random_split
+
+from ..datasets import VHR10
+from ..samplers.utils import _to_tuple
+from ..transforms import AugmentationSequential
+from .geo import NonGeoDataModule
+
+# kornia DataKeys must accept MASKS
+
+
+class _AugPipe(Module):
+    """Pipeline for applying augmentations sequentially on select data keys."""
+
+    def __init__(
+        self, augs: Callable[[Dict[str, Any]], Dict[str, Any]], batch_size: int
+    ) -> None:
+        """Initialize a new _AugPipe instance.
+
+        Args:
+            augs: Augmentations to apply.
+            batch_size: Batch size
+        """
+        super().__init__()
+        self.augs = augs
+        self.batch_size = batch_size
+
+    def forward(self, batch: Dict[str, Tensor]) -> Dict[str, Tensor]:
+        """Apply the augmentation.
+
+        Args:
+            batch: Input batch.
+
+        Returns:
+            Augmented batch.
+        """
+        batch_len = len(batch["image"])
+        for bs in range(batch_len):
+            batch_dict = {
+                "image": batch["image"][bs],
+                "masks": batch["masks"][bs],
+                "labels": batch["labels"][bs],
+                "boxes": batch["boxes"][bs],
+            }
+
+            batch_dict = self.augs(batch_dict)
+
+            batch["image"][bs] = batch_dict["image"]
+            batch["masks"][bs] = batch_dict["masks"]
+            batch["labels"][bs] = batch_dict["labels"]
+            batch["boxes"][bs] = batch_dict["boxes"]
+
+        # Stack images
+        batch["image"] = rearrange(batch["image"], "b () c h w -> b c h w")
+
+        return batch
+
+
+def collate_fn(batch: List[Dict[str, Tensor]]) -> Dict[str, Any]:
+    """Custom object detection collate fn to handle variable boxes.
+
+    Args:
+        batch: list of sample dicts return by dataset
+
+    Returns:
+        batch dict output
+    """
+    output: Dict[str, Any] = {}
+    output["image"] = [sample["image"] for sample in batch]
```

Comment:
Do all keys need to be split, or only boxes?

Reply:
Each image will have N boxes, N labels and N instance segmentation masks. Since N varies image to image, they can't be stacked so they need to be split.

Reply:
How does Kornia normally handle this?

Reply:
From the detection examples I've seen, like the one [here](https://github.com/kornia/tutorials/blob/355d33b068db877beb39f9e6c5866dd6dfa10c96/scripts/training/object_detection/main.py#L74), it's done in a similar manner i.e. via list of dicts. There aren't much (if any) examples on instance segmentation.

For example, in `AugmentationSequential` while list of boxes work, list of masks doesn't. By default the MASK datakey assumes a segmentation style mask tensor.

This is why I had to add MASKS as an alias in `kornia.constants`  so that we could differentiate between mask and masks in our `AugmentationSequential` and handle it appropriately.

Reply:
Can we move this to `.utils` so that vhr10 and nasa_marine_debris can share it? If you think this is common enough, we could even move it into `torchgeo.datasets.utils` and export it like we do with our other custom collation functions.

Reply:
Still need to move